akvo/rsr/management/commands/report_d2d.py

```diff
@@ -6,82 +6,146 @@
         | Q(content_type__model='result')
     )
-    log_entries = LogEntry.objects.filter(action_time__lt=before_date)\
+    log_entries = LogEntry.objects.filter(action_time__range=(date_start, date_end))\
                                   .filter(filter_)\
                                   .values_list('content_type__model', 'object_id')
-
-    indicators = [
-        id_ for _, id_ in
-        filter(lambda x: x[0] == 'indicator', log_entries)
-    ]
     periods = [
         id_ for _, id_ in
         filter(lambda x: x[0] == 'indicatorperiod', log_entries)
     ]
-    results = [
-        id_ for _, id_ in
-        filter(lambda x: x[0] == 'result', log_entries)
-    ]
-
     projects_from_periods = set(
         Project.objects
         .filter(results__indicators__periods__in=periods)
         .values_list('id', flat=True)
     )
+    indicators = [
+        id_ for _, id_ in
+        filter(lambda x: x[0] == 'indicator', log_entries)
+    ]
     projects_from_indicators = set(
         Project.objects
         .filter(results__indicators__in=indicators)
         .values_list('id', flat=True)
     )
+    results = [
+        id_ for _, id_ in
+        filter(lambda x: x[0] == 'result', log_entries)
+    ]
     projects_from_results = set(
         Project.objects.filter(results__in=results)
         .values_list('id', flat=True)
     )
+    return (
+        projects_from_comments
+        | projects_from_updates
+        | projects_from_periods
+        | projects_from_indicators
+        | projects_from_results
+    )
+
+
+def get_unique_logins(date_start, date_end):
+    return LoginLog.objects.\
+        filter(success=True, created_at__range=(date_start, date_end))\
+        .values('email')\
+        .annotate(total=Count('email'))
+
 
-    results_projects = (projects_from_comments
-                        | projects_from_updates
-                        | projects_from_periods
-                        | projects_from_indicators
-                        | projects_from_results)
+def get_published_projects(date_latest):
+    return Project.objects.filter(created_at__lt=date_latest, publishingstatus__status=PublishingStatus.STATUS_PUBLISHED)
 
-    return results_projects
+
+def get_indicator_updates(date_start, date_end):
+    return IndicatorPeriodData.objects\
+        .filter(Q(created_at__range=(date_start, date_end)) | Q(last_modified_at__range=(date_start, date_end)))
+
+
+def get_orgs_reporting_to_iati(date_start, date_end):
+    return IatiExport.objects\
+        .filter(created_at__range=(date_start, date_end))\
+        .annotate(project_count=Count('projects'))\
+        .filter(project_count__gt=0)\
+        .values_list('reporting_organisation_id', flat=True)\
+        .distinct()
+
+
+def get_projects_reporting_to_iati(date_start, date_end):
+    return IatiExport.objects\
+        .filter(created_at__range=(date_start, date_end))\
+        .values_list('projects', flat=True)\
+        .distinct()
```

Comment:
I think these are confusing. The names indicate that they return objects, but one returns IDs and another objects (projects or IDs?). IMO, they should be rewritten to return `QuerySet`s of their types: `Organisation.objects.[...]` and `Project.objects.[...]`.